src/Microsoft.VisualStudio.ProjectSystem.Managed/ProjectSystem/LanguageServices/Handlers/SourceItemHandler.cs

```diff
@@ -167,5 +167,21 @@ private IEnumerable<string> GetFolders(string fullPath, IProjectTreeServiceState
 
             } while (!parent.IsRoot());
         }
+
+        private Task<IProjectTreeServiceState> PublishTreeAsync(IProjectVersionedValue<IProjectSubscriptionUpdate> e)
+        {
+            try
+            {
+                // Make sure the tree matches the same version of the evaluation that we're handling
+                return _projectTree.TreeService.PublishTreeAsync(e.ToRequirements(), blockDuringLoadingTree: true);
+            }
+            catch (ActiveProjectConfigurationChangedException)
+            {
+                // Project configuration changed while we blocked waiting for the tree, instead of trying to match
+                // exactly the same version, just give up and publish the latest tree. Note, don't need to handle
+                // ActiveProjectConfigurationChangedException as it will retry if the configuration changed.
+                return _projectTree.TreeService.PublishLatestTreeAsync(blockDuringLoadingTree: true);
```

Comment:
It won't "match" but it will likely contain the same source files in the tree. If we bail, this will result in us not passing folders to Roslyn.